### PR TITLE
Create CIP

### PR DESCRIPTION
**Pull request by pontusmelke**
A CIP for `CREATE`

### CODE REVIEW

File: cip/CIP2015-10-12-CREATE.adoc [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+toc::[]
+
+== Motivation & Background
+`CREATE` is how nodes and relationships are created in Cypher.
+The syntax has been around since Cypher 1.7 but never properly formalized.
+
+== Proposal
+
+`CREATE` is used to create new nodes and relationships into the graph, as well as set initial labels, type, and properties on the new entities.
```
Comment (petraselmer) [inline cip/CIP2015-10-12-CREATE.adoc]: "create new nodes and relationships into the graph" -> "add new nodes and relationships to the graph"

File: cip/CIP2015-10-12-CREATE.adoc [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Authors:* Pontus Melke pontus.melke@neotechnology.com, Mats Rydberg mats@neotechnology.com
+
+[abstract]
+.Abstract
+--
+This proposal formalizes `CREATE` which is used in Cypher to create nodes and relationships.
```
Comment (nigelsmall) [inline cip/CIP2015-10-12-CREATE.adoc]: Does/should this cover `CREATE UNIQUE` as well (which is part of the same parser rule in the Neo4j implementation)? Similarly, might be also worth mentioning that `CREATE` is used in other contexts within Cypher, such as `CREATE CONSTRAINT`, for clarity.

File: cip/CIP2015-10-12-CREATE.adoc [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+[source, cypher]
+.Create multiple nodes with properties from a parameter list of maps.
+----
+UNWIND { mapList } AS map
+CREATE (n:Node)
+SET n = map
+----
+
+[source, cypher]
+// It seems the . operator doesn't allow multiple lines, and we want two sentences here.
```
Comment (petraselmer) [inline cip/CIP2015-10-12-CREATE.adoc]: I suggest deleting the comment "// It seems...."

File: cip/CIP2015-10-12-CREATE.adoc [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+- `CREATE (a {param})` where `{param}` is a list of maps where one node is created for each item in the list.
+See the example with `UNWIND` above for a way to achieve the same behavior.
+
+- `CREATE (a:L {prop: 42})-[:R]->(b:B)` where both nodes and relationships are created from the same pattern.
+With this proposal one would have to be more specific and write `CREATE (a:L {prop:42}), (b:B), (a)-[:R]->(b)`.
+
+- `CREATE p = (andres { name: 'Andres' })-[:WORKS_AT]->(neo)<-[:WORKS_AT]-(michael { name:'Michael' }) RETURN p`, where a full path is created and returned.
+See the final example above for how to achieve the same behaviour.
+
+We believe that the implicit creation of nodes when specifying a relationship pattern is causing more complexity than whatever benefit it provides to the user, and thus forcing the user to be specific about what to create will benefit both users and implementors of Cypher.
```
Comment (petraselmer) [inline cip/CIP2015-10-12-CREATE.adoc]: "more complexity than whatever benefit it provides to the user" -> "more complexity than any benefit provided to the user"

File: cip/CIP2015-10-12-CREATE.adoc [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+=== Interaction with existing features
+
+`CREATE` on a node pattern will cause the query to fail in the commit phase if the to-be-created node would violate a uniqueness constraint in the database.
+
+This proposal drops some functionality around how we handle creating nodes and relationships:
+
+- `CREATE (a {param})` where `{param}` is a list of maps where one node is created for each item in the list.
+See the example with `UNWIND` above for a way to achieve the same behavior.
+
+- `CREATE (a:L {prop: 42})-[:R]->(b:B)` where both nodes and relationships are created from the same pattern.
```
Comment (nigelsmall) [inline cip/CIP2015-10-12-CREATE.adoc]: If it's only a problem on GitHub, let's not worry too much.

**Comment by systay** [comment]
replaced by #9

File: cip/CIP2015-10-12-CREATE.adoc (hunk 2 above)
Comment (Mats-SX) [inline cip/CIP2015-10-12-CREATE.adoc]: I'll mention that these two things, `CREATE UNIQUE` and `CREATE CONSTRAINT` are _not_ included.

**Pull request closed without merging**